### ISSUE TITLE
Connection descriptor store is not shared across brokers

### ISSUE THREAD

**Issue by josephglanville** [streamnative/mop#57]
The MQTT spec calls for terminating existing clients with the same ClientID on `CONNECT`:
```
If the ClientId represents a Client already connected to the Server then the Server MUST disconnect the existing Client.
```
This is currently implemented using the `ConnectionDescriptorStore` however it's not shared between brokers so connection from a client to another broker within the cluster will violate this part of the spec.

We either need to decide to implement the spec correctly or document non-compliance so users can be aware.

**Comment by codelipenghui** [comment]
@liangyepianzhou This will introduce more workload on the metadata server since the connection can create and close frequently. My opinion is to use one broker to handling the MQTT topic under a namespace, this means all the topics under a namespace are maintained by one broker, so we don't need to share something between MQTT brokers. So that we can guarantee client ID uniqueness under a namespace.

**Comment by josephglanville** [comment]
@codelipenghui this eliminates a large benefit of using Pulsar over a traditional broker which is high availability if this was to be enforced.

At the end of the day MQTT sessions are designed to be long lived. So for well behaved clients it shouldn't be a problem. Could always introduce QoS quotas for number of connection establishments if load on the metadata API is of concern.

**Comment by cbornet** [comment]
Another problem with not sharing and/or storing the session is that the subscription info will be lost if the client connects to another broker or the broker restarts. This is particularly an issue when using `cleanSession=0`. This info could be stored in metadata store and fetched from a local cache for performance reason.

**Comment by Technoboy-** [comment]
> Another problem with not sharing and/or storing the session is that the subscription info will be lost if the client connects to another broker or the broker restarts. This is particularly an issue when using `cleanSession=0`. This info could be stored in metadata store and fetched from a local cache for performance reason.

Hi, we have supported this already.

**Comment by Technoboy-** [comment]
> @codelipenghui this eliminates a large benefit of using Pulsar over a traditional broker which is high availability if this was to be enforced.
> 
> At the end of the day MQTT sessions are designed to be long lived. So for well behaved clients it shouldn't be a problem. Could always introduce QoS quotas for number of connection establishments if load on the metadata API is of concern.

Hi @josephglanville 
As merged #560 and #565, we have supported this in MoP cluster.
